### PR TITLE
Consolidate c7 whats new

### LINKED ISSUE #19971: Combine C# 7.x "what's new" articles
Now that we're approaching .NET 5.0, and the version defaults are C# 7.3 (instead of 7.0), the four articles describing the differences in the 7.x versions should be combined into one article.

Part of #19716

### PR DESCRIPTION
**Pull request by BillWagner**
Fixes #19971 

Since C# 7.3 is the final release supported by .NET framework, it makes sense to consolidate these pages. Furthermore, the changes to the default language version means 7.3 is the default version where it was previously 7.0.

### CODE REVIEW

File: docs/csharp/whats-new/csharp-7.md [hunk 1]
```diff
@@ -1,115 +1,98 @@
...
-ms.date: 02/20/2019
+ms.date: 10/02/2020
 ms.assetid: fd41596d-d0c2-4816-b94d-c4d00a5d0243
 ---
 
-# What's new in C# 7.0
+# What's new in C# 7.0 through C# 7.3
 
-C# 7.0 adds a number of new features to the C# language:
+C# 7.0 through C# 7.3 brought a numberof features and incremental improvements your development experience with C#. This article provides an overview of the new language features and compiler options. The descriptions describe the behavior for C# 7.3, which is the most recent version supported for .NET Framework-based applications.
```
Comment (Youssef1313) [inline docs/csharp/whats-new/csharp-7.md:10]:
```suggestion
C# 7.0 through C# 7.3 brought a number of features and incremental improvements to your development experience with C#. This article provides an overview of the new language features and compiler options. The descriptions describe the behavior for C# 7.3, which is the most recent version supported for .NET Framework-based applications.
```

File: docs/csharp/whats-new/csharp-7.md [hunk 2]
```diff
@@ -1,115 +1,98 @@
...
+
+Finally, the compiler has new options:
 
-The remainder of this article provides an overview of each feature. For each feature,
-you'll learn the reasoning behind it. You'll learn the syntax. You can explore these features in your environment using the `dotnet try` global tool:
+- `-refout` and `-refonly` that control [reference assembly generation](#reference-assembly-generation).
+- `-publicsign` to enable Open Source Software (OSS) signing of assemblies.
+- `-pathmap` to provide a mapping for source directories.
+
+The remainder of this article provides an overview of each feature. For each feature, you'll learn the reasoning behind it. You'll learn the syntax. You can explore these features in your environment using the `dotnet try` global tool:
```
Comment (Youssef1313) [inline docs/csharp/whats-new/csharp-7.md:50]:
```suggestion
The remainder of this article provides an overview of each feature. For each feature, you'll learn the reasoning behind it and the syntax. You can explore these features in your environment using the `dotnet try` global tool:
```

**Review by CamSoper — APPROVED** [review]

**Comment by gewarren** [comment]
@mavasani pointed out that there were warnings introduced with this PR in the redirection file. @BillWagner will you be able to fix these?

**Comment by BillWagner** [comment]
@gewarren Yes, I'll fix those today.

**Pull request merged**